packages/clerk-js/src/ui/elements/PasswordInput.tsx

```diff
@@ -1,34 +1,110 @@
 import { EyeSlash } from '../icons';
 import { useFormControl } from '../primitives/hooks';
 import type { PropsOfComponent } from '../styledSystem';
 import { IconButton } from './IconButton';
 
 type PasswordInputProps = PropsOfComponent<typeof Input> & {
+  strengthMeter?: boolean;
   complexity?: boolean;
 };
 
+/**
+ * Coordinate the success and error states between password complexity and strength
+ * `usePasswordStrength` and `usePasswordComplexity` will fire callbacks to handle error state for their own
+ * cases.
+ * This hook aims to synchronize those states and throw 1 error if such exists
+ * - Complexity errors has priority over strength errors.
+ * - To set a field as successful, both complexity and strength need to be successful
+ */
+const useComplexityOverStrength = ({ strengthMeter }: Required<Pick<PasswordInputProps, 'strengthMeter'>>) => {
+  const formControlProps = useFormControl();
+  const hasComplexityError = useRef(false);
+  const hasComplexitySuccess = useRef(false);
+  return {
+    setStrengthSuccess: useCallback(() => {
+      if (!hasComplexityError.current) {
+        formControlProps.setSuccessful?.(true);
+      }
+    }, []),
+
+    setStrengthError: useCallback((errorMessage: string) => {
+      if (!hasComplexityError.current) {
+        formControlProps.setError?.(errorMessage);
+      }
+    }, []),
+
+    setComplexityError: useCallback((errorMessage: string) => {
+      hasComplexityError.current = !!errorMessage;
+      formControlProps.setError?.(errorMessage);
+    }, []),
+
+    setComplexitySuccess: useCallback(() => {
+      hasComplexityError.current = false;
+      hasComplexitySuccess.current = true;
+      if (!strengthMeter) {
+        formControlProps.setSuccessful?.(true);
+      }
+    }, []),
+  };
+};
+
+const loadZxcvbn = () => {
+  return Promise.all([
+    // @ts-ignore
+    import(/*webpackIgnore: true*/ 'https://cdn.jsdelivr.net/npm/@zxcvbn-ts/core@2.2.1/+esm'),
+    // @ts-ignore
+    import(/*webpackIgnore: true*/ 'https://cdn.jsdelivr.net/npm/@zxcvbn-ts/language-common@2.0.1/+esm'),
+  ]).then(([core, zxcvbnCommonPackage]) => {
+    core.zxcvbnOptions.setOptions({
+      dictionary: {
+        ...zxcvbnCommonPackage.default.dictionary,
+      },
+      graphs: zxcvbnCommonPackage.default.adjacencyGraphs,
+    });
+    return core.zxcvbn;
+  });
+};
+
 export const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>((props, ref) => {
   const [hidden, setHidden] = React.useState(true);
-  const { id, onChange: onChangeProp, complexity = false, ...rest } = props;
-  const formControlProps = useFormControl();
+  const { id, onChange: onChangeProp, strengthMeter = false, complexity = false, ...rest } = props;
   const {
     userSettings: { passwordSettings },
   } = useEnvironment();
 
+  const { show_zxcvbn } = passwordSettings;
+
+  const { setStrengthError, setComplexityError, setStrengthSuccess, setComplexitySuccess } = useComplexityOverStrength({
```

Comment:
❓ How come we need 3 separate hooks here? I wonder if the orchestration you need here would become simpler we combined everything into a single hook, avoiding the callbacks. 

Eg:
```tsx
const { password, setPassword, score } = usePassword(settings);
```  

`setPassword` would take care of the orchestration internally

Reply:
@nikosdouvlis That was one way to go, which crossed my mind. But I was told that we may end up shipping this functionality as part of the public API, and I wanted to keep things separate. 

Do you propose merging usePasswordComplexity and usePasswordStrength

or create a new hook (`usePassword`) that uses both under the hood and does the coordination internally ?

Reply:
Would the public hooks need this kind of orchestration?
To be honest, I'd go with the simpler approach for now (usePassword) as I'd like us to start with a design document for any additions to the public api :)
Also  note that this helper is within clerk-js - it cannot be exposed as a toplevel api from clerk/clerk-react directly so we'll probably have to reimplement it at some point.

Reply:
TBH I don't have an idea of what will be the intended use of the public hook(s).

I'll have another look, on how `usePassword` could improve what we have now